### ISSUE TITLE
Tundra not registering setup opts

### ISSUE THREAD

**Issue by chris-av** [sam4llis/nvim-tundra#11]
Hey there! I really love this colorscheme. Was wondering if I am importing this colorscheme incorrectly? I copied and pasted the code from the README. However, I am getting the following error: 

![Screen Shot 2023-07-06 at 5 04 09 PM](https://github.com/sam4llis/nvim-tundra/assets/41182258/a2478a8d-def8-4519-b3b6-42df677294cb)

I always get this error. Doesn't matter if I paste the defaults into the setup function or an empty table.

The only way to fix this error, is if I code into the repo itself and manually call the `M.setup` within the `M.load` function. like so:

```lua
-- nvim-tundra/lua/nvim-tundra/init.lua
M.load = function()
  local opts = vim.g.tundra_opts or base_configuration
  M.setup() -- added this line
... 

}
```

Am I importing the colorscheme incorrectly? Using it just like this (copied and pasted the config from README): 

```lua
-- ~/.config/nvim/lua/pks/nvim-tundra
require('nvim-tundra').setup({
  transparent_background = false,
  dim_inactive_windows = {
    enabled = false,
    color = nil,
  },
  sidebars = {
    enabled = true,
    color = nil,
  },
  editor = {
    search = {},
    substitute = {},
  },
  syntax = {
    booleans = { bold = true, italic = true },
    comments = { bold = true, italic = true },
    conditionals = {},
    constants = { bold = true },
    fields = {},
    functions = {},
    keywords = {},
    loops = {},
    numbers = { bold = true },
    operators = { bold = true },
    punctuation = {},
    strings = {},
    types = { italic = true },
  },
  diagnostics = {
    errors = {},
    warnings = {},
    information = {},
    hints = {},
  },
  plugins = {
    lsp = true,
    semantic_tokens = true,
    treesitter = true,
    telescope = true,
    nvimtree = true,
    cmp = true,
    context = true,
    dbui = true,
    gitsigns = true,
    neogit = true,
    textfsm = true,
  },
  overwrite = {
    colors = {},
    highlights = {},
  },
})
```

**Comment by sam4llis** [comment]
Hi thanks for creating an issue. Haven’t seen this issue before but will try investigating it this week (I will be abroad though so can’t promise this) :)

Just to rule out the obvious, you are definitely calling `:colorscheme tundra` after the setup function and getting this error?

**Comment by chris-av** [comment]
Yup! It actually doesn't seem to affect the error in any way whether I call the below code after the setup function :

```lua
vim.opt.background = 'dark'
vim.cmd('colorscheme tundra')
```

I just updated my neovim config. You can see how I am calling it [here](https://github.com/chris-av/nvim/blob/main/lua/pkgs/nvim-tundra.lua). Hope you're having fun abroad! Thanks again for such a wonderful color scheme. One of my fav's. 😎

**Comment by sam4llis** [comment]
Thanks for the information :) Will have a look at this as soon as I can!

**Comment by sam4llis** [comment]
Once again thanks for the details above, really helped with debugging! I've added a fix commit to the `dev` branch (9ca3744) which should allow you to use the Tundra [base configuration](https://github.com/sam4llis/nvim-tundra/blob/main/lua/nvim-tundra/base_configuration.lua) without using any `setup` function in your personal configuration. Meaning it should work with *only* these lines:

```lua
vim.opt.background = 'dark'
vim.cmd('colorscheme tundra')
```

If you could check that it works for you that would be great, and if all works I will close the issue and create a `v0.4.1` on the `main` branch with this fix :) Thanks!

**Comment by chris-av** [comment]
Hey there @sam4llis I just checked out the dev branch and it works for me. Thanks! :pray: 

I've also tried the the setup function and also that works too! I've managed to successfully customize a highlight color. So both solutions appear to work now!

![swappy-20230716_153359](https://github.com/sam4llis/nvim-tundra/assets/41182258/67055c17-df3a-46dc-8b09-5e1bab2c5af3)


Thanks again! You're the best.
